libraries/AP_HAL_Linux/OpticalFlow_Onboard.cpp

```diff
@@ -179,13 +204,49 @@ void OpticalFlow_Onboard::_run_optflow()
     Vector3f gyro_rate;
     Vector2f flow_rate;
     VideoIn::Frame video_frame;
+    uint32_t convert_buffer_size, crop_buffer_size;
+    uint32_t crop_width, crop_height;
+    uint8_t *convert_buffer, *crop_buffer;
     uint8_t qual;
 
+    if (_format == V4L2_PIX_FMT_YUYV) {
+        convert_buffer_size = _crop_by_software ? _crop_by_software_width * _crop_by_software_height : _width * _height;
+        convert_buffer = (uint8_t *) alloca(convert_buffer_size);
+    }
+
+    if (_crop_by_software) {
+        crop_buffer_size = HAL_OPTFLOW_ONBOARD_OUTPUT_WIDTH * HAL_OPTFLOW_ONBOARD_OUTPUT_HEIGHT;
+        crop_buffer = (uint8_t *) alloca(crop_buffer_size);
+        crop_width = (_crop_by_software_width - HAL_OPTFLOW_ONBOARD_OUTPUT_WIDTH) / 2;
+        crop_height = (_crop_by_software_height - HAL_OPTFLOW_ONBOARD_OUTPUT_HEIGHT) / 2;
+    }
+
     while(true) {
         /* wait for next frame to come */
         if (!_videoin->get_frame(video_frame)) {
             AP_HAL::panic("OpticalFlow_Onboard: couldn't get frame\n");
         }
+
+        if (_format == V4L2_PIX_FMT_YUYV) {
+            VideoIn::convert_from_yuyv_to_grey((uint8_t *)video_frame.data, video_frame.bytesused, convert_buffer);
+
+            memset(video_frame.data, 0, video_frame.bytesused);
+            memcpy(video_frame.data, convert_buffer, convert_buffer_size);
+            video_frame.bytesused = convert_buffer_size;
+        }
+
+        if (_crop_by_software) {
+            if (_format != V4L2_PIX_FMT_YUYV && _format != V4L2_PIX_FMT_GREY) {
+                AP_HAL::panic("OpticalFlow_Onboard: software crop not supported\n");
+            }
+
+            VideoIn::crop_image_grey((uint8_t *)video_frame.data, video_frame.bytesused, crop_buffer, _crop_by_software_width, _crop_by_software_height, crop_width, crop_width, crop_height, crop_height);
```

Comment:
Could you please limit the number of columns. I don't think there is a limitation in the ardupilot coding style but I would appreciate it to remain less than 80 characters, or maybe a little more but not more than 100 please.